sbin/sriov-setmac.sh

```
@@ -0,0 +1,199 @@
+#
+
+BDF_REGEX="^[[:xdigit:]]{2}:[[:xdigit:]]{2}.[[:xdigit:]]$"
+DBDF_REGEX="^[[:xdigit:]]{4}:[[:xdigit:]]{2}:[[:xdigit:]]{2}.[[:xdigit:]]$"
+VD_REGEX="^[[:xdigit:]]{4}:[[:xdigit:]]{4}$"
+
+if [[ $EUID -ne 0 ]]; then
+    echo "Error: This script must be run as root" 1>&2
+    exit 1
+fi
+
+if [[ -z "$@" ]]; then
```

Shellcheck (0.11.0)

[error] Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).

(SC2199)